lib/events.js

```diff
@@ -435,16 +450,62 @@ function listenerCount(type) {
 }
 
 EventEmitter.prototype.eventNames = function eventNames() {
-  return this._eventsCount > 0 ? Reflect.ownKeys(this._events) : [];
+  const count = this._eventsCount;
+  if (count === 0)
+    return [];
+
+  const events = this._events;
+  const actualEventNames = new Array(count);
+  var j = 0;
+  for (var key in events) {
+    if (events[key] !== undefined) {
+      actualEventNames[j] = key;
+      j++;
+    }
+  }
+
+  // We must have Symbols to fill in
+  if (j < count) {
```

Comment:
Tiny nit: It would make sense to me if the comment was inside the `if` since it refers to what must be true if the condition is fulfilled... does that make sense?